packages/appstore/src/components/main-title-bar/__tests__/asset-summary.spec.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import AssetSummary from '../asset-summary';
+import { render, screen } from '@testing-library/react';
+import { StoreProvider, mockStore } from '@deriv/stores';
+import { isMobile } from '@deriv/shared';
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isMobile: jest.fn(() => false),
+}));
+
+jest.mock('Components/pre-loader/total-assets-loader', () => ({
+    __esModule: true,
+    default: () => <div>TotalAssetLoader</div>,
+}));
+
+jest.mock('@deriv/components', () => ({
+    ...jest.requireActual('@deriv/components'),
+    Popover: jest.fn(props => (
+        <>
+            <span>{props.message}</span>
+            <span>{props.alignment}</span>
+        </>
+    )),
+}));
+
+let mock = mockStore({});
+
+const checkContainerAssetSummary = () => {
+    const wrapper = ({ children }: { children: JSX.Element }) => <StoreProvider store={mock}>{children}</StoreProvider>;
+
+    const { container } = render(<AssetSummary />, {
+        wrapper,
+    });
+    expect(container).toBeInTheDocument();
+};
+
+describe('AssetSummary', () => {
+    beforeEach(() => {
+        mock = mockStore({});
+    });
+    it('should render asset summary', () => {
+        checkContainerAssetSummary();
+    });
+
+    it('should render asset summary with total asset', () => {
+        checkContainerAssetSummary();
+        expect(screen.getByText('Total asset')).toBeInTheDocument();
+    });
+
+    it('should not render the total asset text if isMobile is true', () => {
+        isMobile.mockImplementation(() => true);
+        checkContainerAssetSummary();
+        expect(screen.queryByText('Total asset')).not.toBeInTheDocument();
+    });
+
+    it('should render the TotalAssetsLoader component if is_switching is true and user has eu_account ', () => {
+        mock.traders_hub.selected_account_type = 'real';
+        mock.traders_hub.is_eu_user = true;
+        mock.client.is_switching = true;
+
+        checkContainerAssetSummary();
```

Comment:
@thisyahlen-deriv Please don't do this 🙏🏻 

Each test case should be isolated and not rely on something global as changing that thing will affect the test, That's why we have created the `mockStore` to override the values you need and get a constant in each test case.